pyramid_swagger/tween.py

```diff
@@ -93,6 +99,28 @@ def _get_validation_context(registry):
         return noop_context
 
 
+def get_swagger_handler(settings, route_info, registry):
+    """This returns back `settings.swagger_handler` in almost of the cases.
+
+    Exception is made when `settings.include_2dot0_routes` are non-empty and
+    ['1.2', '2.0'] both are present in available swagger versions. In this
+    special scenario, '2.0' spec is chosen only for requests which are listed
+    in the `include_2dot0_routes`. This helps in incremental migration of
+    routes from v1.2 to v2.0 by making moving to v2.0 opt-in.
+
+    :rtype: :class:`SwaggerHandler`
+    """
+    enabled_swagger_versions = get_swagger_versions(registry.settings)
+    if (SWAGGER_20 in enabled_swagger_versions
+        and SWAGGER_12 in enabled_swagger_versions
+        and settings.include_2dot0_routes
+        and route_info.get('route')
+            and route_info['route'].name not in settings.include_2dot0_routes):
+        return settings.swagger12_handler
+
+    return settings.swagger_handler
```

Comment:
I'd vote for code simplicity over reducing the number of if statements per call. I bet our jsonschema serde cost swamps it :)